client/components/editor/teaching-elements/Assessment/TextSelection/Selections.js

```diff
@@ -0,0 +1,32 @@
+import min from 'lodash/min';
+import max from 'lodash/max';
+import Range from './Range';
+import reject from 'lodash/reject';
+
+export default class Selections {
+  constructor(selections = []) {
+    this.ranges = selections.map(it => new Range(...it));
+  }
+
+  merge(range) {
+    const overlapses = this.ranges.filter(it => it.intersectsWith(range));
+    range.start = min([...overlapses.map(it => it.start), range.start]);
+    range.end = max([...overlapses.map(it => it.end), range.end]);
```

Comment:
I'm not sure if you aware of it but `Math.max` & `Math.min` are actually capable of accepting `...args` 😉